### PR TITLE
Fixed leaking color syntax from Jira to Slack

### PR DESCRIPTION
**Pull request by Diana520**
Valid Jira text formatting syntax is rendered correctly in Jira, but is shown in raw form in Slack notifications. If text was coloured, Slack added color tags to comment. To fix this bug, i used regex to get the message without tags.

### CODE REVIEW

File: jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CommentUtilTest {
+
+    @Test
+    void testRemoveJiraTagsRemovesValidTags() {
+        assertEquals("Some text with a title",
```
Comment (utluiz) [inline jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java:12]: Multiple similar assertions are a good candidate to "table tests" or "parameterized tests". See this documentation, for instance: https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource

File: jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                CommentUtil.removeJiraTags("{quote}here is quotable content to be quoted{q1uote}"));
+        assertEquals("{noformat:}preformatted piece",
+                CommentUtil.removeJiraTags("{noformat:}preformatted piece{noformat}"));
+        assertEquals("some text{coode}",
+                CommentUtil.removeJiraTags("{code:title=Bar.java|borderStyle=solid}some text{coode}"));
+        assertEquals("{notsupperted}message{notsupported}",
+                CommentUtil.removeJiraTags("{notsupperted}message{notsupported}"));
+        assertEquals("", CommentUtil.removeJiraTags(null));
+    }
+
```
Comment (utluiz) [inline jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java:41]: Regex can sometimes be dangerous. Depending on a combination of pattern and input, the algorithm might enter into an exponential computation and freeze the whole process. Example: https://www.regular-expressions.info/catastrophic.html

I think it would be important testing some specific character combinations to make sure the Regexes behave as expected under uncommon conditions. Also, even if the current regexes are safe, testing corner cases will prevent issues in future changes.

Some examples I can think of:
- `{{{{{{{{{{{{{{{{{{{{{{{{color}}}}}}}}}}}}}}}}}}}}}}}}}`
- `{quote{quote{quote{quote{quote{quote{quote{quote{quote}}}}}}`

**Review by utluiz — APPROVED** [review]

File: jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+class CommentUtilTest {
+    @ParameterizedTest
+    @MethodSource("validTagsProvider")
+    void testRemoveJiraTagsRemovesValidTags(String argument) {
+        assertEquals("some text", CommentUtil.removeJiraTags(argument));
+    }
+
+    static Stream<String> validTagsProvider() {
+        return Stream.of("{panel:title=My Title}some text{panel}",
```
Comment (utluiz) [inline jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java:21]: FWIW you could have used a Stream with two values (original and expected) so you would have just one dataset and there would be no need for hard coding the correct value.

An instance from [Stack Overflow](https://stackoverflow.com/questions/61483452/parameterized-test-with-two-arguments-in-junit-5-jupiter):

```

@ParameterizedTest
@MethodSource("provideParameters")
public void testParametersFromMethod(String name, int value) {
    System.out.println("method data " + name + " value " + value);
}

private static Stream<Arguments> provideParameters() {
    return Stream.of(
            Arguments.of("a", 1),
            Arguments.of("b", 2),
            Arguments.of("foo", 3)
    );
}
```

File: jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java (hunk 1 above)
Comment (Diana520) [reply jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java]: I added parameterized tests instead of simple tests

File: jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java (hunk 2 above)
Comment (Diana520) [reply jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java:50]: I added separate test for checking these complex inputs that pass fast enough

**Review by mvlasovatl — APPROVED** [review]

File: jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java (hunk 3 above)
Comment (Diana520) [reply jira-slack-server-integration/jira-slack-server-integration-plugin/src/test/java/com/atlassian/jira/plugins/slack/util/CommentUtilTest.java:21]: Nice idea, i think i will use it next time
